utest/parsing/test_statements.py

```diff
@@ -56,11 +57,21 @@ def assert_created_statement(tokens, base_class, **params):
     )
 
 
-def assert_statements(first, second):
-    for t1, t2 in zip(first, second):
+def compare_statements(first, second):
+    return (isinstance(first, type(second))
+            and first.tokens == second.tokens
+            and first.errors == second.errors)
+
+
+def assert_statements(st1, st2):
+    for t1, t2 in zip(st1, st2):
         assert_equal(t1, t2, formatter=repr)
-    assert_equal(first, second)
-    assert_equal(len(first), len(second))
+    assert_true(
+        compare_statements(st1, st2),
+        'Statements are not equal. %s (%s) != %s (%s)' % (st1, type_name(st1),
```

Comment:
I could also import report_inequality from robot.asserts OR create new method inside robot asserts that will check if given condition is true.